### ISSUE TITLE
Test failure when font names contain only numeric characters

### ISSUE THREAD

**Issue by charlesej** [pygame/pygame#740]
The following fails are seen when running the pygame tests (`python -m pygame.tests`).

### Fail details: ###
Fail from pygame 1.9.4.
```
======================================================================
FAIL: test_get_fonts (pygame.tests.font_test.FontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Program Files\Python37\lib\site-packages\pygame\tests\font_test.py", line 99, in test_get_fonts
    self.failUnless(name.islower(), name)
AssertionError: False is not true : 3270

----------------------------------------------------------------------
Ran 712 tests in 104.630s

FAILED (failures=1)
```

Fails from pygame 1.9.5.dev0 (886b3dd288d7953db123d6ab1cca895653cc3eea). (Note: file paths edited).
```
======================================================================
FAIL: test_get_fonts (pygame.tests.font_test.FontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "\pygame\tests\font_test.py", line 94, in test_get_fonts
    self.assertTrue(name.islower(), name)
AssertionError: False is not true : 3270

======================================================================
FAIL: test_get_fonts (pygame.tests.ftfont_test.FtFontModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "\pygame\tests\font_test.py", line 94, in test_get_fonts
    self.assertTrue(name.islower(), name)
AssertionError: False is not true : 3270

----------------------------------------------------------------------
Ran 848 tests in 75.238s

FAILED (failures=2)
```

### System details: ###
- os: windows 10 (64bit)
- python: 3.7.2 (64bit)
- pygame: 1.9.4 and 1.9.5.dev0 at 886b3dd288d7953db123d6ab1cca895653cc3eea
- numpy: 1.15.4

The following fonts exist. 
```
C:\>dir /b Windows\Fonts\3*
3270Medium.ttf
3270Medium_0.ttf
3270Narrow.ttf
3270Narrow_0.ttf
3270SemiNarrow.ttf
3270SemiNarrow_0.ttf
```

When setting up the available fonts on this system, pygame parses the 3270Narrow fonts to a font name of 3270. This code shows that the 3270 font exists and is different than the default system font.
```
import pygame
import time

def create_text(font_name):
    if font_name is None:
        # Hack to show where freesansbold.ttf is coming from.
        font_path = pygame.font.match_font('freesans')
        font_path += ' ({})'.format(pygame.font.get_default_font())
    else:
        font_path = pygame.font.match_font(font_name)

    text = '{}: {}'.format(font_name, font_path)
    font = pygame.font.SysFont(font_name, 30)
    print(text)

    return font.render(text, True, pygame.Color('white'))


pygame.init()
screen = pygame.display.set_mode((640, 100))

name_3270 = '3270'
print([f for f in pygame.font.get_fonts() if f.startswith(name_3270)])

screen.blit(create_text(None), (0, 0))
screen.blit(create_text(name_3270), (0, 50))

pygame.display.flip()
time.sleep(3)
pygame.quit()
```

The above code prints this output to the terminal. In addition, the images in the pygame window are visually different.
```
['3270medium', '3270', '3270seminarrow']
None: C:\WINDOWS\Fonts\arial.ttf (freesansbold.ttf)
3270: C:\WINDOWS\Fonts\3270Narrow_0.ttf
```

### Debugging Details: ###
The test failures occur on this [line of code](https://github.com/pygame/pygame/blob/886b3dd288d7953db123d6ab1cca895653cc3eea/test/font_test.py#L94) when checking all the font names on a system.
```
self.assertTrue(name.islower(), name)
```
The failing font name ('3270') is all numeric characters. The `.islower()` will return `False` because it needs at least one **cased** character for it to return `True`.

From the [python docs](https://docs.python.org/3/library/stdtypes.html#str.islower).
> str.islower()
> Return true if all cased characters [4] in the string are lowercase and there is at least one cased character, false otherwise.

**TLDR:** On my system the 3270 (IBM 3270) font exists and is usable. The pygame tests erroneously fail due to the parsed font name being entirely numeric.
